server/protocols/webadmin/src/test/java/org/apache/james/webadmin/routes/UsersRoutesTest.java

```diff
@@ -0,0 +1,397 @@
+
+import static com.jayway.restassured.RestAssured.given;
+import static com.jayway.restassured.RestAssured.when;
+import static com.jayway.restassured.RestAssured.with;
+import static com.jayway.restassured.config.EncoderConfig.encoderConfig;
+import static com.jayway.restassured.config.RestAssuredConfig.newConfig;
+import static org.apache.james.webadmin.Constants.SEPARATOR;
+import static org.apache.james.webadmin.WebAdminServer.NO_CONFIGURATION;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.equalTo;
+import static org.hamcrest.CoreMatchers.is;
+import static org.mockito.Matchers.any;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import org.apache.james.domainlist.api.DomainList;
+import org.apache.james.user.api.UsersRepository;
+import org.apache.james.user.api.UsersRepositoryException;
+import org.apache.james.user.api.model.User;
+import org.apache.james.user.memory.MemoryUsersRepository;
+import org.apache.james.webadmin.WebAdminServer;
+import org.apache.james.webadmin.utils.JsonTransformer;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.google.common.base.Charsets;
+import com.jayway.restassured.RestAssured;
+import com.jayway.restassured.parsing.Parser;
+
+import de.bechte.junit.runners.context.HierarchicalContextRunner;
+
+@RunWith(HierarchicalContextRunner.class)
+public class UsersRoutesTest {
+
+    public static final String DOMAIN = "domain";
+    public static final String USERNAME = "username@" + DOMAIN;
+    public static final String PATH_SPECIFIC_USER = UserRoutes.USERS + SEPARATOR + USERNAME;
+    private WebAdminServer webAdminServer;
+
+    private void createServer(UsersRepository usersRepository) throws Exception {
+        webAdminServer = new WebAdminServer(new UserRoutes(usersRepository, new JsonTransformer()));
+        webAdminServer.configure(NO_CONFIGURATION);
+        webAdminServer.await();
+
+        RestAssured.port = webAdminServer.getPort();
+        RestAssured.config = newConfig().encoderConfig(encoderConfig().defaultContentCharset(Charsets.UTF_8));
+        RestAssured.defaultParser = Parser.JSON;
+    }
+
+    @After
+    public void stop() {
+        webAdminServer.destroy();
+    }
+
+    public class NormalBehaviour {
+
+        @Before
+        public void setUp() throws Exception {
+            DomainList domainList = mock(DomainList.class);
+            when(domainList.containsDomain(DOMAIN)).thenReturn(true);
+
+            MemoryUsersRepository usersRepository = MemoryUsersRepository.withVirtualHosting();
+            usersRepository.setDomainList(domainList);
+
+            createServer(usersRepository);
+        }
+
+
+        @Test
+        public void getUsersShouldBeEmptyByDefault() {
+            when()
+                .get(UserRoutes.USERS)
```

Comment:
Again, please be more explicit !
I missunderstood this comment and work for 10 minutes for nothing.
I would suggest something like : "Restassured allows for a baseURI, avoiding you to repete yourself"

Not enough detailled review comments leads to misunderstanding (and thus useless works), demands interpretetion (and thus takes time), and is frustrating. I think you could easily take a couple of seconds to write more readable review comments. I don't see why being readable would apply just to code!

And here, I will make you this comment on the next JMAP review (you do not use it)